### PR TITLE
End-To-End tests of download project/directory/file feature

### PR DESCRIPTION
**Pull request by dmytro-ndp**
### What does this PR do?
It adds _End-To-End_ tests of download project/directory/file feature of Eclipse Che.

### What issues does this PR fix or reference?
#9538

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
ci-test

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java [hunk 1]
```diff
@@ -44,4 +48,25 @@ public boolean isCheRunLocally() {
 
     return false;
   }
+
+  public void copy(String containerId, Path pathInsideContainer, Path copyTo) throws IOException {
```
Comment (musienko-maxim) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java]: Need to have JavaDoc here.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java [hunk 2]
```diff
@@ -44,4 +48,25 @@ public boolean isCheRunLocally() {
...
+    String copyCommand =
+        format("docker cp %1$s:%2$s %3$s", containerId, pathInsideContainer, copyTo);
+
+    processAgent.execute(copyCommand);
+  }
+
+  public String findGridNodeContainerByIp(String Ip) throws ProcessAgentException {
```
Comment (musienko-maxim) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java]: Need to have JavaDoc here.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java [hunk 3]
```diff
@@ -44,4 +48,25 @@ public boolean isCheRunLocally() {
...
+    String getContainerIdCommand =
+        format(
+            "docker ps -q --filter='name=selenium_chromenode*' | xargs docker inspect --format '{{ .Id }} {{ .NetworkSettings.Networks.selenium_selenium_grid_internal.IPAddress }}' | grep %s | awk 'NR>0 {print $1;}'",
+            Ip);
+    return processAgent.execute(getContainerIdCommand);
+  }
+
+  public void delete(String gridNodeContainerId, Path pathToDelete) throws ProcessAgentException {
```
Comment (musienko-maxim) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/DockerUtil.java]: Need to have JavaDoc here.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/DownloadedIntoGridFileUtilImpl.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.UnauthorizedException;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.utils.DockerUtil;
+
+/**
+ * This is set of methods to work with files which are downloaded into WebDriver.
```
Comment (musienko-maxim) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/DownloadedIntoGridFileUtilImpl.java]: Please, make the JavaDoc more specific.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/DownloadedLocallyFileUtilImpl.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+
+/**
+ * This is set of methods to work with files which are downloaded into WebDriver.
```
Comment (musienko-maxim) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/DownloadedLocallyFileUtilImpl.java]: Please, make the JavaDoc more specific.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+
+    // when
+    projectExplorer.openContextMenuByPathSelectedItem(TEST_PROJECT_1);
+    projectExplorer.clickOnItemInContextMenu(ContextMenuFirstLevelItems.DOWNLOAD);
+
+    // then
+    assertEquals(
+        getPackageFileList(DOWNLOADED_TEST_PROJECT_1_PACKAGE_NAME),
+        expectedPackageFileList.toString());
+
```
Comment (musienko-maxim) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java:155]: need to remove downloaded package here

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+    projectExplorer.clickOnItemInContextMenu(ContextMenuFirstLevelItems.DOWNLOAD);
+
+    // then
+    String downloadedFileContent = getDownloadedFileContent(TEST_FILE_NAME);
+    assertEquals(downloadedFileContent, testFileContent);
+  }
+
+  private String getDownloadedFileContent(String testFileName) throws IOException {
+    try {
+      return downloadedFileUtil.getDownloadedFileContent(
```
Comment (musienko-maxim) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java]: Let's improve waiting on download - do it up to 5 times with 1 second period.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+    } catch (IOException e) {
+      WaitUtils.sleepQuietly(2);
+      return downloadedFileUtil.getDownloadedFileContent(
+          seleniumWebDriver.getSessionId(), testFileName);
+    }
+  }
+
+  private String getPackageFileList(String downloadedTestProject1PackageName) throws IOException {
+    try {
+      return downloadedFileUtil
```
Comment (musienko-maxim) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/DownloadProjectTest.java]: Let's improve waiting on download - do it up to 5 times with 1 second period.

**Review by musienko-maxim — APPROVED** [review]

**Pull request merged**